### PR TITLE
Add support for building CIRCT from source

### PR DESCRIPTION
**Pull request by jerryz123**
Unfortunately, some machines we use cannot run the CIRCT nightlies, with no workaround besides building CIRCT from source. 
This adds a script and flag to build-setup to build CIRCT from source.

**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [ ] New feature
- [ ] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [ ] Other

**Contributor Checklist**:
- [ ] Did you set `main` as the base branch?
- [ ] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [ ] Did you state the type-of-change/impact?
- [ ] Did you delete any extraneous prints/debugging code?
- [ ] Did you mark the PR with a `changelog:` label?
- [ ] (If applicable) Did you update the conda `.conda-lock.yml` file if you updated the conda requirements file?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you add a test demonstrating the PR?

- [ ] (If applicable) Did you mark the PR as `Please Backport`?

### CODE REVIEW

**Review by joonho3020 — APPROVED** [review]
LGTM

File: scripts/build-setup.sh [hunk 1]
```diff
@@ -306,13 +310,20 @@ if run_step "10"; then
...
-    git submodule update --init $CYDIR/tools/install-circt &&
-    $CYDIR/tools/install-circt/bin/download-release-or-nightly-circt.sh \
-        -f circt-full-static-linux-x64.tar.gz \
-        -i $PREFIX \
-        -v version-file \
-        -x $CYDIR/conda-reqs/circt.json \
-        -g null
+    if [ "$BUILD_CIRCT" = true ] ; then
+	echo "Building CIRCT from source, and installing to $PREFIX"
+	$CYDIR/scripts/build-circt-from-source --prefix $PREFIX
```
Comment (abejgonzalez) [inline scripts/build-setup.sh:315]:
```suggestion
	$CYDIR/scripts/build-circt-from-source.sh --prefix $PREFIX
```

**Review by abejgonzalez — APPROVED** [review]
I think it would be a good idea to change our full flow CI (which is used for CAD stuff + other) to use this flag (just to cover that the build works).

**Comment by jerryz123** [comment]
> I think it would be a good idea to change our full flow CI (which is used for CAD stuff + other) to use this flag (just to cover that the build works).

This does massively blow up the run time of that flow though.... I'm actually ok with not testing this in CI... its for one very specific environment. Very few people should use this.

**Pull request merged**